src/integrationTest/java/com/mongodb/kafka/connect/ChangeStreamRoundTripTest.java

```diff
@@ -0,0 +1,317 @@
+    sourceProperties.put(FULL_DOCUMENT_CONFIG, "updateLookup");
+    sourceProperties.put(COLLECTION_CONFIG, coll1.getNamespace().getCollectionName());
+    addSourceConnector(sourceProperties);
+
+    Properties sinkProperties = getSinkProperties(replicated, coll1);
+    sinkProperties.put(COLLECTION_CONFIG, coll1.getNamespace().getCollectionName());
+    addSinkConnector(sinkProperties);
+
+    assertDatabase(original, replicated);
+
+    // Test can handle multiple collections, with inserts and updates
+    insertMany(rangeClosed(100, 110), coll1);
+    coll1.updateMany(new Document(), Updates.set("test", 1));
+    assertDatabase(original, replicated);
+
+    // Test can handle replace and update operations
+    insertMany(rangeClosed(120, 130), coll1);
+    coll1.replaceOne(eq("_id", 1), new Document());
+    coll1.replaceOne(eq("_id", 3), new Document());
+    coll1.replaceOne(eq("_id", 5), new Document());
+
+    assertDatabase(original, replicated);
+
+    // Test handles delete operations
+    insertMany(rangeClosed(140, 150), coll1);
+    coll1.deleteMany(Filters.mod("_id", 2, 0));
+
+    assertDatabase(original, replicated);
+  }
+
+  @Test
+  @DisplayName("Ensure database CRUD operations can be round tripped")
+  void testRoundTripDatabaseCrud() {
+    assumeTrue(isGreaterThanThreeDotSix());
+    MongoDatabase original = getDatabaseWithPostfix();
+    MongoDatabase replicated = getDatabaseWithPostfix();
+    MongoCollection<Document> coll1 = original.getCollection("coll1");
+    MongoCollection<Document> coll2 = original.getCollection("coll2");
+
+    // Test existing messages will be round tripped
+    insertMany(rangeClosed(1, 10), coll1);
+
+    Properties sourceProperties = getSourceProperties(original);
+    sourceProperties.put(FULL_DOCUMENT_CONFIG, "updateLookup");
+    addSourceConnector(sourceProperties);
+    addSinkConnector(getSinkProperties(replicated, coll1, coll2));
+
+    assertDatabase(original, replicated);
+
+    // Test can handle multiple collections, with inserts and updates
+    insertMany(rangeClosed(100, 110), coll1, coll2);
+    coll1.updateMany(new Document(), Updates.set("test", 1));
+    assertDatabase(original, replicated);
+
+    // Test can handle replace and update operations
+    insertMany(rangeClosed(120, 130), coll1, coll2);
+    coll1.replaceOne(eq("_id", 1), new Document());
+    coll1.replaceOne(eq("_id", 3), new Document());
+    coll1.replaceOne(eq("_id", 5), new Document());
+    coll2.updateMany(new Document(), Updates.set("newField", 1));
+
+    assertDatabase(original, replicated);
+
+    // Test handles delete operations
+    insertMany(rangeClosed(140, 150), coll1, coll2);
+    coll1.deleteMany(Filters.mod("_id", 2, 0));
+    coll2.deleteMany(Filters.mod("_id", 3, 0));
+
+    assertDatabase(original, replicated);
+  }
+
+  @Test
+  @DisplayName("Ensure collection CRUD operations are replicated")
+  void testPipelineBasedUpdatesCanBeRoundTripped() {
```

Comment:
Here updates are specifically tested to ensure that truncatedArrays support is added and works. This uses the test cases from the server js tests. It also ensures that the databases are as expected at various points.